outbound.js

```diff
@@ -1492,7 +1514,12 @@ HMailItem.prototype.try_deliver_host_on_socket = function (mx, host, port, socke
         return send_command('MAIL', 'FROM:' + self.todo.mail_from);
     };
 
+    var fp_called = false;
```

Comment:
It's to ensure that `finish_processing_mail` doesn't get called twice. It's standard practise to use a closure like this.